File: Java_ChildrenCompetition/src/main/java/repository/AbstractRepository.java
package repository;

import java.util.HashMap;
import java.util.Map;

public abstract class AbstractRepository<ID, T extends HasId<ID>> implements IRepository<ID, T> {
    protected Map<ID, T> entities;
    protected Validator<T> validator;

    public AbstractRepository(Validator<T> validator) {
        this.entities = new HashMap<>();
        this.validator = validator;
    }

    @Override
    public int size() {
        return 0;
    }

    @Override
    public void update(ID id, T entity) { }

    @Override
    public void save(T entity) { }

    @Override
    public void delete(ID id) { }

    @Override
    public T findOne(ID id) {
        return null;
    }

    @Override
    public Iterable<T> findAll() {
        return null;
    }
}

File: CSharp_ChildrenCompetition/CSharp_ChildrenCompetition/CSharp_ChildrenCompetition/repository/AbstractRepository.cs
using System.Collections.Generic;

namespace CSharp_ChildrenCompetition.repository
{
    public class AbstractRepository<ID, T> : IRepository<ID, T> where T : HasId<ID>
    {
        protected Dictionary<ID, T> entities;
        protected Validator<T> validator;

        public AbstractRepository(Validator<T> validator)
        {
            this.entities = new Dictionary<ID, T>();
            this.validator = validator;
        }

        public int size()
        {
            return 0;
        }

        public void save(T entity)
        {
            
        }

        public void delete(ID id)
        {
            
        }

        public void update(ID id, T entity)
        {
            
        }

        public T findOne(ID id)
        {
            return default;
        }

        public IEnumerable<T> findAll()
        {
            return default;
        }

    }
}